apps/prairielearn/src/middlewares/authzCourseOrInstance.js

```diff
@@ -185,20 +207,6 @@ module.exports = asyncHandler(async (req, res, next) => {
   let is_administrator = res.locals.is_administrator;
   let user_with_requested_uid_has_instructor_access_to_course_instance = false;
 
-  // If the user is not enrolled in the course instance but could be, then
-  // automatically enroll them.
-  if (
-    isCourseInstance &&
-    res.locals.authz_data.authn_has_student_access &&
-    !res.locals.authz_data.authn_has_student_access_with_enrollment
-  ) {
-    // Enroll authenticated user in course instance
-    await sqldb.queryAsync(sql.ensure_enrollment, {
-      course_instance_id: res.locals.course_instance.id,
-      user_id: user.user_id,
-    });
-  }
-
```

Comment:
This auto-enrollment check was almost certainly redundant, as the `authz_course_instance` sproc used in the `select_authz_data` query would have already enrolled the user if it was possible to do so.